STAT/STATLinkDef.h

```diff
@@ -28,6 +28,13 @@
 #pragma link C++ class AliTreeTrending+;
 #pragma link C++ class AliNDFormulaBrowser+;
 #pragma link C++ class AliDrawStyle+;
+#pragma link C++ function <Int_t> AliDrawStyle::GetNamedTypeAt(const char *, Bool_t &, Int_t , const char *, Int_t, const char , const char *);
```

Comment:
Hi Ruben, Do you suggest to don't use template and use explicit type e.g. Float_t for this function and then use cast, because and you are absolutely right, I know where I need Int_t or Float_t?

But my idea was to provide mechanism for more explicit cast for users, so In your case if I will look to the list of functions in cint I will see just Float_t MyFunc, but I prefer see such list: <Int_t> MyFucnt(...), <Float_t> MyFunc(...), but in any case for cint such links doesn't work and user see nothing, it works only for cling without links because I use relevant callings in my code.

thank you for your comment, we will think and perhaps we will take your advice.

Reply:
By the way, if you know how to add link for templated output functions, I will be happy to know about it too :)

Reply:
@bdrum my point was that your GetNamedTypeAt contains lines like
````
if (...) return <float>;
else return <int>;
````
With your template implementation the compiler will pick either float or int specialization according to LHS, and this specialization of GetNamedTypeAt will internally cast one of these return types to the type of LHS. But it will do exactly the same also w/o template: the cast will be done in all cases.

For the LinkDef, I think you should use 
````
#pragma link C++ function AliDrawStyle::GetNamedTypeAt<int>
#pragma link C++ function AliDrawStyle::GetNamedTypeAt<float>
````

Anyway, this is a minor problem. Currently the test fails on compilation with root6.

Reply:
Hallo Boris.

I remember you were running both root5 and root6. Could you confirm, resp. repeat root6 test.
This time  I did not run root 6 test. Eventually I can recompile mu root 6.

Marian

Reply:
@shahor02,  I agree, I will change it in the future commits, it really doesn't have a sense, thanks! 
And thanks for your LinkDef strings, it works.

@miranov25 Hello, Marian.
Unfortunately I have used root 6 only for compiling my classes, but not whole project.